src/Aspire.Hosting.Yarp/api/Aspire.Hosting.Yarp.cs

```diff
@@ -13,6 +13,8 @@ public partial interface IYarpConfigurationBuilder
         Yarp.YarpCluster AddCluster(ApplicationModel.EndpointReference endpoint);
         Yarp.YarpCluster AddCluster(ApplicationModel.IResourceBuilder<ExternalServiceResource> externalService);
         Yarp.YarpCluster AddCluster(ApplicationModel.IResourceBuilder<IResourceWithServiceDiscovery> resource);
+        Yarp.YarpCluster AddCluster(string clusterName, object destination);
```

Comment:
Did we ever get around to adding some type-checks for this? Maybe we should have multiple overloads like we do for other APIs?

Reply:
I opened an issue to track this, since it isn't trivial. https://github.com/dotnet/aspire/issues/11521